x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml

```diff
@@ -158,6 +164,141 @@
             "analysis": {"outlier_detection":{}}
           }
 
+---
+"Test put config with invalid dest index name":
+
+  - do:
+      catch: /Invalid index name \[<script>Foo\]*/
+      ml.put_data_frame_analytics:
+        id: "config-with-invalid-dest-index"
+        body: >
+          {
+            "source": {
+              "index": "index-source"
+            },
+            "dest": {
+              "index": "<script>Foo"
```

Comment:
It might be worth adding one more test, where the destination index name is a valid wildcard, e.g. `mydest*`, to prove that wildcarded destinations are not allowed.